bitnami/nginx/templates/networkpolicy.yaml

```diff
@@ -0,0 +1,70 @@
+{{- /*
+Copyright VMware, Inc.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if .Values.networkPolicy.enabled }}
+kind: NetworkPolicy
+apiVersion: {{ include "common.capabilities.networkPolicy.apiVersion" . }}
+metadata:
+  name: {{ template "common.names.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  {{- $podLabels := include "common.tplvalues.merge" ( dict "values" ( list .Values.podLabels .Values.commonLabels ) "context" . ) }}
+  podSelector:
+    matchLabels: {{- include "common.labels.matchLabels" ( dict "customLabels" $podLabels "context" $ ) | nindent 6 }}
+  policyTypes:
+    - Ingress
+    - Egress
+  egress:
+    {{- if .Values.networkPolicy.allowExternalEgress }}
+    - ports:
+        - {}
+    {{- else }}
+    - ports:
+        # Allow dns resolution
+        - port: 53
+          protocol: UDP
+        - port: 53
+          protocol: TCP
+    {{- if .Values.networkPolicy.extraEgress }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.networkPolicy.extraEgress "context" $ ) | nindent 4 }}
+    {{- end }}
+    {{- end }}
+  ingress:
+    - ports:
+        - port: {{ .Values.containerPorts.http }}
+        - port: {{ .Values.containerPorts.https }}
```

Comment:
This here requires a default value.
https://github.com/bitnami/charts/blob/16358846b22bbfaf9a8cc01ae1d462a07d33ccd0/bitnami/nginx/values.yaml#L257

I believe kubernetes will reject an empty string but I might be wrong.
I just updated to 15.10.1 and kubeconform rejected this chart due to 


> not valid: problem validating schema. Check JSON formatting: jsonschema: '/spec/ingress/0/ports/1/port' does not validate with v1.28.3-standalone-strict/networkpolicy-networking-v1.json#/properties/spec/properties/ingress/items/properties/ports/items/properties/port/oneOf: valid against schemas at indexes 0 and 1

Reply:
https://github.com/bitnami/charts/issues/23058